tokio-macros/src/entry.rs

```diff
@@ -380,7 +380,64 @@ fn parse_knobs(mut input: ItemFn, is_test: bool, config: FinalConfig) -> TokenSt
         }
     };
 
-    let body = input.body();
+    let unit_type = Box::new(parse_quote! { () });
+    let never_type = Box::new(parse_quote! { ! });
+
+    let output_type = match &input.sig.output {
+        // For functions with no return value syn doesn't print anything,
+        // but that doesn't work as `Output` for our boxed `Future`, so
+        // default to `()` (the same type as the function output).
+        syn::ReturnType::Default => &unit_type,
+        syn::ReturnType::Type(_, ret_type) if ret_type == &never_type => &unit_type,
+        syn::ReturnType::Type(_, ret_type) => ret_type,
+    };
+
+    enum ContainsLoopVisitor {
+        Found,
+        NotFound,
+    }
+
+    impl syn::visit::Visit<'_> for ContainsLoopVisitor {
+        fn visit_expr_loop(&mut self, _: &syn::ExprLoop) {
+            *self = ContainsLoopVisitor::Found;
+        }
```

Comment:
This is definitely super wasteful to parse the entire syntax tree until hitting a loop. 
`Visit` has 182 methods and I could add `skip!` to all of them if `*self == Found` but that's a lof of nothing code just to fix an error message